dxp/build_process.md

```diff
@@ -0,0 +1,67 @@
+
+![An overview of the build](../images/liferay-npm-scripts-pipeline.png)
+
+When running `liferay-npm-scripts build` a sequence of execution of other tools takes place, depending on the environment in which it is being run, will trigger other tools (e.g. Webpack and Soy).
+
+1. Setup env
+2. Validation of the `npmscripts.config.js` file if it exists
+3. Build [Soy](http://metaljs.com/docs/guides/soy-components.html#soy_compilation) files, if any
+4. Build [Babel](https://babeljs.io/) transformation
+5. Run the [Webpack](https://webpack.js.org/) if the `webpack.config.js` file exists
+6. Run [Liferay npm bundler](https://github.com/liferay/liferay-js-toolkit/tree/master/packages/liferay-npm-bundler)
+7. Run the [Liferay npm bridge generator](https://github.com/liferay/liferay-js-toolkit/tree/master/packages/liferay-npm-bridge-generator) if it exists `.npmbridgerc`
+
+CLI does a lot for you, like [not having to configure babel, eslint, jest, npm-bundler, prettier and stylelint](https://github.com/liferay/liferay-npm-tools/tree/master/packages/liferay-npm-scripts/src/config), you don't need to configure this when using liferay-npm-scripts, you also overwrite these settings by adding the config file of the respective tools, the CLI will take your settings and apply the merge.
+
+### Build Pipeline
+
+Following the sequence we can describe in more detail what happens in each phase briefly.
+
+By default the CLI expects its source code to be in `src/main/resources/META-INF/resources` and is compiled into the `build/node/packageRunBuild/resources` folder.
+
+#### 1. Setup env
+
+Only set `NODE_ENV` to `production`, you can easily overwrite it just by setting `env NODE_ENV=development liferay-npm-scripts build`.
+
+#### 2. Validate `npmscripts.config.js`
+
+The CLI will validate the `npmscripts.config.js` configuration file looking for required keys, just to avoid errors that are difficult to track later.
+
+> You can read more about the [npmscripts.config.js settings](https://github.com/liferay/liferay-npm-tools/tree/master/packages/liferay-npm-scripts#config) in the [liferay-npm-tools](https://github.com/liferay/liferay-npm-tools) repository.
+
+#### 3. Soy
+
+If there is a `.soy` file in the module, `liferay-npm-scripts` will build Soy and also Babel it next and drop it into the build destination folder. Unlike the other phases, before executing the babel the Soy build will compile your `.soy` code and drop it into a temporary folder (by default in `build/npmscripts`) until the babel can transform and send it to destination folder.
+
+The CLI will run the [`metalsoy`](https://github.com/metal/metal-tools-soy/tree/4.x) tool to handle with the Soy files.
+
+#### 4. Babel
+
+As part of the build process, [babel](https://babeljs.io/) is used for transformation, for React with JSX syntax for example and to allow the use of new JavaScript features.
```

Comment:
Maybe mention what "level" of JavaScript we are configuring. Or at least provide a link to the configuration file to be able to see which plugins are added easily.

Reply:
Yeah, I think I’ll just link it up, maybe things will change and we have to keep the list updated.